median.dfy

```diff
@@ -0,0 +1,56 @@
+module Std.Median {
+    
+    import Std.Sort; //This will work when we have a sorting library which will have different sorting techniques
```

Comment:
As I mentioned in the sync I do think it's smart to make use of the existing sequence-based mergesort initially. But on second thought it would be nice to have the more efficient array-based mergesort in the standard libraries (or ANY array-based sort method really), so it would be nice to add this too as a stretch goal!